src/services/confirmation.statement.service.ts

```diff
@@ -1,11 +1,22 @@
 import { Session } from "@companieshouse/node-session-handler";
-import { ConfirmationStatementService } from "private-api-sdk-node/dist/services/confirmation-statement";
+import {
+  CompanyValidationResponse,
+  ConfirmationStatementCreated,
+  ConfirmationStatementService
+} from "private-api-sdk-node/dist/services/confirmation-statement";
+import Resource, { ApiErrorResponse } from "@companieshouse/api-sdk-node/dist/services/resource";
 import { createPrivateOAuthApiClient } from "./api.service";
 
 export const createConfirmationStatement = async (session: Session,
-                                                  transactionId: string): Promise<any> => {
+                                                  transactionId: string): Promise<Resource<ConfirmationStatementCreated | CompanyValidationResponse>> => {
   const client = createPrivateOAuthApiClient(session);
   const csService: ConfirmationStatementService = client.confirmationStatementService;
-  return await csService.postNewConfirmationStatement(transactionId);
+  const response = await csService.postNewConfirmationStatement(transactionId);
+  if (response.httpStatusCode !== 201 && response.httpStatusCode !== 400) {
+    const castedResponse: ApiErrorResponse = response;
+    throw new Error("Something went wrong creating confirmation statement " + JSON.stringify(castedResponse));
```

Comment:
might have the same issue here. Digging into it more, the ApiErrorResponse is a CH defined interface so 'hopefully' should not have circular refs, it is usually more the system interfaces that do.
But, I'd at least change the string to use placeholders using the back ticks and ${} 
`    throw new Error("Something went wrong creating confirmation statement ${JSON.stringify(castedResponse)}");
`
replace quotes with backticks